code/game/machinery/flatpacker.dm

```diff
@@ -0,0 +1,344 @@
+			return ITEM_INTERACT_BLOCKING
+		// If insertion was successful and there's already a diskette in the console, eject the old one.
+		if(inserted_board)
+			inserted_board.forceMove(drop_location())
+
+		inserted_board = attacking_item
+		// 5 sheets of iron and 5 of cable coil
+		needed_mats = list()
+		for(var/type as anything in inserted_board.req_components)
+			needed_mats = analyze_cost(type, needed_mats)
+
+		CREATE_AND_INCREMENT(needed_mats, /datum/material/iron, (SHEET_MATERIAL_AMOUNT * 5 + (SHEET_MATERIAL_AMOUNT / 20)))
+		CREATE_AND_INCREMENT(needed_mats, /datum/material/glass, (SHEET_MATERIAL_AMOUNT / 20))
+
+		update_appearance()
+		return ITEM_INTERACT_SUCCESS
+
+	return NONE
+
+/**
+ * Attempts to find the total material cost of a typepath (including our creation efficiency), modifying a list
+ * The list is modified as an assoc list: Material datum typepath = Cost
+ * If the type is found on a techweb, uses material costs from there
+ * Otherwise, the typepath is created in nullspace and fetches materials from the initialized one, then deleted.
+ *
+ * Args:
+ * type - Typepath of the item we are trying to find the costs of
+ * costs - Assoc list we modify and return
+ */
+/obj/machinery/flatpacker/proc/analyze_cost(type, costs)
+	var/comp_type = type
+	if(ispath(type, /datum/stock_part))
+		var/datum/stock_part/as_part = type
+		comp_type = initial(as_part.physical_object_type)
+		if(as_part.tier > print_tier)
+			print_tier = as_part.tier
+
+	var/by_techweb = !isnull(SSresearch.item_to_design[comp_type])
+	var/obj/item/null_comp = by_techweb ? null : new comp_type
+	var/list/mat_list = by_techweb ? SSresearch.item_to_design[comp_type][1].materials : null_comp.custom_materials
+	for(var/atom/mat as anything in mat_list)
+		var/mat_type = mat.type
+
+		CREATE_AND_INCREMENT(costs, mat_type, mat_list[mat] * inserted_board.req_components[type])
+
+	qdel(null_comp)
+	return costs
+
+/// Start building the currently inserted board, if possible
+/obj/machinery/flatpacker/proc/start_build()
+	. = FALSE
+	if(!inserted_board)
+		return
+	if(!materials.has_materials(needed_mats, creation_efficiency))
+		say("Not enough materials to begin production.")
+		return
+	if(print_tier > max_part_tier)
+		say("Design too complex.")
+		return
+	materials.use_materials(needed_mats, creation_efficiency)
+	playsound(src, 'sound/items/rped.ogg', 50, TRUE)
+	busy = TRUE
+
+	addtimer(CALLBACK(src, PROC_REF(finish_build), inserted_board), flatpack_time)
+	return TRUE
+
+/// turns the supplied board into a flatpack, and sets the machine as not busy
+/obj/machinery/flatpacker/proc/finish_build(board)
+	busy = FALSE
+	new /obj/item/flatpack(drop_location(), board)
+
+/obj/machinery/flatpacker/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(gone == inserted_board)
+		inserted_board = null
+		needed_mats = null
+		print_tier = 1
+		update_appearance()
+
+/obj/machinery/flatpacker/ui_act(action, list/params)
+	. = ..()
+
+	if(.)
+		return
+
+	. = TRUE
```

Comment:
```suggestion
```

Don't force the return value if we haven't handled it yet. returning true always will update the UI needlessly